packer/plugin-getter/plugins.go

```diff
@@ -210,7 +210,14 @@ func (l InstallList) String() string {
 // comparing the version lexicographically.
 // A Duplicate version will replace any already present version.
 func (l *InstallList) InsertSortedUniq(install *Installation) {
-	pos := sort.Search(len(*l), func(i int) bool { return (*l)[i].Version >= install.Version })
+	pos := sort.Search(len(*l), func(i int) bool {
+		elem := (*l)[i]
+		if elem.BinaryPath == install.Version {
```

Comment:
How is elem.BinaryPath ever going to equal install.Version? This feels like it will always be false.

Reply:
Good catch, both the comparisons failed us in this case, and the check after to determine whether to skip or not the inclusion of the binary did not check that the paths were the same, hence why we didn't have them set in the right order.
That being said, I think we can get rid of the whole insertion function here, and directly add plugin installs as we find them, the glob already produces lexicographically ordered entries, and if the version is not the only determining factor here, and we consider two binaries with a different identifier with the same version as separate (which I think we should, even if the binary is fundamentally equal), we could remove this logic completely.